pkg/apis/apis/v1alpha1/api_cel_test.go

```diff
@@ -0,0 +1,216 @@
+			validBinding: false,
+		},
+		{
+			name: "invalid non core resource",
+			claim: map[string]interface{}{
+				"group":    "new.core.resources",
+				"resource": "fakeresources",
+			},
+			validBinding: false,
+		},
+		{
+			name: "valid non core resource",
+			claim: map[string]interface{}{
+				"group":        "new.core.resources",
+				"resource":     "fakeresources",
+				"identityHash": "fakehashhere",
+			},
+			validBinding: true,
+		},
+	}
+
+	validator, s := getValidatorForPermissionClaim(t, "../../../../config/crds/apis.kcp.dev_apiexports.yaml", "permissionClaims")
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			errs := validator.Validate(field.NewPath("root"), &s, tc.claim)
+			if len(errs) == 0 && !tc.validBinding {
+				t.Error("No errors were found, but should be invalid binding")
+				return
+			}
+			if len(errs) > 0 && tc.validBinding {
+				t.Errorf("found errors: %v but should be valid binding", errs.ToAggregate().Error())
+				return
+			}
+		})
+	}
+}
+
+func withRule(s schema.Structural, rule string) schema.Structural {
+	s.Extensions.XValidations = apiextensionsv1.ValidationRules{
+		{
+			Rule: rule,
+		},
+	}
+	return s
+}
+
+func primitiveType(typ, format string) schema.Structural {
+	result := schema.Structural{
+		Generic: schema.Generic{
+			Type: typ,
+		},
+	}
+	if len(format) != 0 {
+		result.ValueValidation = &schema.ValueValidation{
+			Format: format,
+		}
+	}
+	return result
+}
+
+func getValidatorForPermissionClaim(t *testing.T, crdFilePath string, specClaimName string) (*cel.Validator, schema.Structural) {
+
+	// Get the file and marhsel to unstructed
+	data, err := os.ReadFile(crdFilePath)
+	require.NoError(t, err)
+	u := &unstructured.Unstructured{}
+	yaml.Unmarshal(data, u)
+
+	versions, found, err := unstructured.NestedSlice(u.Object, "spec", "versions")
+	require.NoError(t, err)
+	require.True(t, found)
+	// TODO: When adding more version, we will need to write a test to cover each version.
+	// The fact that this will fail, will remind us.
+	require.Len(t, versions, 1)
+
+	version, ok := versions[0].(map[string]interface{})
+	require.True(t, ok)
+	validations, found, err := unstructured.NestedSlice(version, "schema", "openAPIV3Schema", "properties", "spec", "properties", specClaimName, "items", "x-kubernetes-validations")
+	require.NoError(t, err)
+	require.True(t, found)
+
+	// We will need to enhance the test, to handle the different rules if any are added. this failure will alert us to that case.
+	require.Len(t, validations, 1)
+
+	validationMap, ok := validations[0].(map[string]interface{})
+	require.True(t, ok)
+	ruleString, ok := validationMap["rule"].(string)
+	require.True(t, ok)
+
+	s := schema.Structural{
+		Generic: schema.Generic{
+			Type: "object",
+		},
+		Properties: map[string]schema.Structural{
+			"group":        primitiveType("string", ""),
+			"resource":     primitiveType("string", ""),
+			"identityHash": primitiveType("string", ""),
+		},
+	}
```

Comment:
This schema can be derived, some random example:
```
	openapiSchema, err := apiResourceVersion.GetSchema()
	if err != nil {
		return nil, err
	}
	if err := apiextensionsv1.Convert_v1_JSONSchemaProps_To_apiextensions_JSONSchemaProps(openapiSchema, internalSchema, nil); err != nil {
		return nil, fmt.Errorf("failed converting CRD validation to internal version: %w", err)
	}
	structuralSchema, err := structuralschema.NewStructural(internalSchema)
```

Reply:
I am sorry this code block is not clear to me what needs to be done.

PermissionClaim does not have the GetSchema() call

Reply:
same for this will be useful for a more generic getting of schemas and CELs. Will add it as a follow-up when needed.